src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/integration/cas2/Cas2DomainEventTest.kt

```diff
@@ -0,0 +1,84 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.integration.cas2
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.events.cas2.model.Cas2ApplicationSubmittedEvent
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.events.cas2.model.EventType
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.factory.events.cas2.Cas2ApplicationSubmittedEventDetailsFactory
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.integration.IntegrationTestBase
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.DomainEventType
+import java.time.Instant
+import java.util.UUID
+
+class Cas2DomainEventTest : IntegrationTestBase() {
+  @Test
+  fun `Get CAS2 Application Submitted Event without JWT returns 401`() {
+    webTestClient.get()
+      .uri("/events/cas2/application-submitted/e4b004f8-bdb2-4bf6-9958-db602be71ed3")
+      .exchange()
+      .expectStatus()
+      .isUnauthorized
+  }
+
+  @Test
+  fun `Get CAS2 Application Submitted Event without ROLE_CAS2_EVENTS returns 403`() {
+    val jwt = jwtAuthHelper.createClientCredentialsJwt(
+      username = "username",
```

Comment:
do you think it would be clearer if you set the `roles` explicitly, even better to a real role which is not permitted access? e.g.

```kt
    val jwt = jwtAuthHelper.createClientCredentialsJwt(
      username = "username",
      roles = listOf("ROLE_APPROVED_PREMISES_EVENTS"),
    )
```

Reply:
great idea!